tests/test_utils.py

```diff
@@ -407,7 +408,8 @@ def get_dataset(
         if not create_engine:
             return None
 
-        engine = create_engine("mysql+pymysql://root@localhost/test_ci")
+        # engine = create_engine("mysql+pymysql://root@localhost/test_ci")
+        engine = create_engine("mysql+pymysql://root@host.docker.internal/test_ci")
```

Comment:
Just at first glance while scrolling through this, I believe this should also be `{db_hostname}`? Spotted this in a few other places in this PR too. I haven't looked too deeply yet, so feel free to clarify if I'm mistaken!

Reply:
Oh Noooo! Sorry for that. You are completely right :/
I will fix that and double-check before marking it.

Reply:
Hi Sam,
I did finish to replace my (supposedly temporary) fix with this simple solution using environment variables. Sorry for that again.

I didn't want to impact too much the code, and this is the solution I found to run/pass most of the tests.
I still get some errors though, but when I tried to solve them on my side, they crashed on Azure devops.
I would gladly have a chat with you and your team if you start investigating these few incompatibilities.
At least, for now, I can run the tests dealing with SqlAlchemy! :) 

Cheers!
Patrick